### PR TITLE
Implemented Collection::chunk()

### PR DESCRIPTION
**Pull request by lorenzo**
This method returns smaller arrays of the given size

### CODE REVIEW

**Comment by josegonzalez** [comment]
Yes yes yes, so much win.

File: src/Collection/CollectionTrait.php [hunk 1]
```diff
@@ -605,6 +605,26 @@ public function zipWith($items, $callable)
...
+            $values = [$v];
+            for ($i = 1; $i < $chunkSize; $i++) {
+                $iterator->next();
+                if (!$iterator->valid()) {
+                    break;
+                }
+                $values[] = $iterator->current();
+            }
+
+            return empty($values) ? $values : [$values];
```
Comment (markstory) [inline src/Collection/CollectionTrait.php]: When would values be empty, if it always contains at least one element from line 611?

File: tests/TestCase/Collection/CollectionTest.php [hunk 2]
```diff
@@ -1527,4 +1527,30 @@ public function testSerializeWithZipIterator()
...
+
+    /**
+     * Tests the chunk method with overflowing chunk size
+     *
+     * @return void
+     */
+    public function testChunkOverflow()
+    {
+        $collection = new Collection(range(1, 11));
+        $chunked = $collection->chunk(2)->toList();
```
Comment (markstory) [inline tests/TestCase/Collection/CollectionTest.php]: What about chunking an iterator that contains nested data? Something like:

```php
$data = [1, 2, 3, [4, 5], 6, [7, [8, 9], 10], 11];
```

Comment (lorenzo) [inline tests/TestCase/Collection/CollectionTest.php]: I can add a test for that.

File: src/Collection/CollectionTrait.php (hunk 1 above)
Comment (lorenzo) [inline src/Collection/CollectionTrait.php]: Hmmm, you're right. This is a leftover of a refactoring I made

**Comment by lorenzo** [comment]
@markstory added the test for the nested elements case and fixed the extra code.

**Pull request merged**